### PR TITLE
[CARBONDATA-1401] fix duplicate issue in ListInfo

### PR DESCRIPTION
**Pull request by lionelcao**
add validation for listInfo

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
SDV Build Success with Spark 2.1, Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/277/

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/CommonUtil.scala [hunk 1]
```diff
@@ -369,6 +369,14 @@ object CommonUtil {
     }
   }
 
+  def validateListInfo(listInfo: List[List[String]]): Unit = {
```
Comment (xuchuanyin) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/CommonUtil.scala]: simplify this by
```
val list = listInfo.flatted
if (list.length != list.distinct.size) {
  sys.error("xxx")
}
``

Comment (xuchuanyin) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/CommonUtil.scala]: two more things:
1. this function is only used once, so I think there is no need to place it in `CommonUtil`.
2. is `distinct` case-sensitive or not?

Comment (lionelcao) [reply integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/CommonUtil.scala]: Hi @xuchuanyin ,
Thank you for the comments. Have changed to use flatten function.
1. Put in CommonUtil not only for reusing but also keep code in one style and put similar function in one place.
2. distinct is case sensitive

**Comment by ravipesala** [comment]
@lionelcao Please add a testcase for this scenario

**Comment by lionelcao** [comment]
@ravipesala Test case added，please verify.

**Comment by ravipesala** [comment]
LGTM

**Comment by ravipesala** [comment]
I have merged but still not closed, Please close it manually

**Comment by CarbonDataQA** [comment]
SDV Build Success with Spark 2.1, Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/325/

**Pull request closed without merging**